docs/data/sdks/sdk-architecture.md

```diff
@@ -0,0 +1,263 @@
+    // Implementation for the debug method
+    console.log(...args);
+  }
+}
+
+amplitude.init('API_KEY', 'OPTIONAL_USER_ID', {
+  loggerProvider: new MyLogger(),
+});
+```
+
+### Event 
+
+Events represent how users interact with your application. 
+
+`BaseEvent` represents a basic event with optional properties. You can also track other event properties in the field of `event_properties`.
+
+```typescript
+// Latest Browser SDK example: track an event
+
+import * as amplitude from '@amplitude/analytics-browser';
+import {BaseEvent} from '@amplitude/analytics-types';
+
+const buttonClickedEvent: BaseEvent = {
+  event_type: 'Button Clicked',
+  event_properties: {
+    buttonColor: 'primary',
+  }
+}
+
+amplitude.track(buttonClickedEvent);
+```
+
+Some SDKs also allow you to track an event without creating one but just pass the event type as a string.
+
+```typescript
+// Latest Browser SDK example: track an event without creating one 
+
+import * as amplitude from '@amplitude/analytics-browser';
+
+// Track a basic event
+amplitude.track('Button Clicked');
+
+// Track events with optional properties
+const eventProperties = {
+  buttonColor: 'primary',
+};
+amplitude.track('Button Clicked', eventProperties);
+```
+
+### Plugin
+
+Plugins allow you to extend Amplitude SDK's behavior. Please check [here](../../sdk-plugin) for more details. 
+
+## Common methods
+
+Latest SDKs share the same architecture as well as the same interfaces. Aligned interfaces benefit using Amplitude SDKs on multiple platforms without memorizing different function names.
+
+This a list of shared interfaces of latest SDKs. 
+
+!!!note 
+    Different SDKs follow different naming conventions. Here use camel case for methods and parameters and use pascal case for classes. 
+
+- `init(apiKey)`
+- `track(BaseEvent)`
+- `identify(IdentifyEvent)`
+- `setGroup(groupType, groupName)`
+- `groupIdentify(groupType, groupName, IdentifyEvent)`
+- `revenue(RevenueEvent)`
+- `flush()`
+- `add(Plugin)`
+- `remove(Plugin)`
+- `shutdown()`
+
+## Comparison with maintenance SDK
+
+If you are migrating from maintenance SDKs, you may notice that latest SDKs differ from maintenance SDKs not only in terms of interfaces but also in their Middleware vs Plugin architecture . You can think of Middleware as being equal to Enrichment Plugin and Destination Plugin.
+
+## SDK status table
+
+|Platform|Follow the latest architecture|Not follow the latest architecture|
+|--------|--------|-----------|
+|Browser|:octicons-package-16: `@amplitude/analytics-browser`<br/>[:material-github: Amplitude-TypeScript](https://github.com/amplitude/Amplitude-TypeScript)<br/>[:material-file-document: Document](./typescript-browser/index.md)|:material-hammer-screwdriver:{.red} Maintenance SDK<br/>:octicons-package-16: `@amplitude/amplitude-js`<br/>[:material-github: Amplitude-JavaScript](https://github.com/amplitude/Amplitude-JavaScript)<br/>[:material-file-document: Document](./javascript/index.md)|
```

Comment:
I think it would be good to have the packages link to releases page for that package e.g. ":octicons-package-16: `@amplitude/amplitude-js`" would liknk to JS SDK releases.